File: petcare_planner_backend/models/Pet.js
const mongoose = require("mongoose");

const PetSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    petImage: {
      type: String,
      required: true,
    },
    petType: {
      type: String,
      enum: ["Dog", "Cat", "Other"],
      required: true,
    },
    petName: {
      type: String,
      required: true,
    },
    breed: {
      type: String,
      required: true,
    },
    age: {
      type: Number,
      required: true,
    },
    gender: {
      type: String,
      enum: ["Male", "Female"],
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("Pet", PetSchema);

File: petcare_planner_frontend/lib/models/pet.dart
class Pet {
  final String id;
  final String petType;
  final String petName;
  final String breed;
  final int age;
  final String gender;
  final String? petImage;

  Pet({
    required this.id,
    required this.petType,
    required this.petName,
    required this.breed,
    required this.age,
    required this.gender,
    this.petImage,
  });

  factory Pet.fromJson(Map<String, dynamic> json) {
    return Pet(
      id: json['_id'],
      petType: json['petType'],
      petName: json['petName'],
      breed: json['breed'],
      age: json['age'],
      gender: json['gender'],
      petImage: json['petImage'],
    );
  }
}
